index.js

```diff
@@ -26,19 +29,30 @@ module.exports = function (src, dest, opts) {
 
 	opts = objectAssign({overwrite: true}, opts);
 
-	return new Promise(function startRead(resolve, reject) {
-		var read = fs.createReadStream(src);
-
-		read.on('error', function (err) {
-			reject(new CpFileError('cannot read from `' + src + '`: ' + err.message, err));
-		});
+	var progress = {
+		stat: null,
+		emitter: new EventEmitter()
+	};
```

Comment:
@schnittstabil while this is nice, I have question if we want to optimize for this at all, is 1 `stat` call worth additional branching and checking? I'm ok with either answer, just wanted to raise a question. Even if it is, we might be able to avoid it in some other way.